vaadin-icons-flow-parent/vaadin-icons-flow/src/main/java/com/vaadin/flow/component/icon/AbstractIcon.java

```diff
@@ -35,7 +35,7 @@
 @NpmPackage(value = "@vaadin/icon", version = "24.2.0-alpha11")
 @JsModule("@vaadin/icon/src/vaadin-icon.js")
 public abstract class AbstractIcon extends Component
-        implements ClickNotifier<Icon>, HasTooltip {
+        implements ClickNotifier<AbstractIcon>, HasTooltip {
```

Comment:
<img width="678" alt="Screenshot 2023-08-29 at 14 25 39" src="https://github.com/vaadin/flow-components/assets/1222264/7a48e9ae-be7a-4de0-a4dd-df3bea76cdd8">

I think the `getSource()` should not return `AbstractIcon`, but in this case, it should be an `Icon`.

Reply:
Should AbstractIcon be changed to `AbstractIcon<T extends AbstractIcon<T>>` allowing to pass the proper generic to the ClickNotifier?

Reply:
I added a generic type to `AbstractIcon` to pass it to `ClickNotifier`, and had to change the implementation in `Icon`  to pass itself as the param type.